Update following work with app-manifest at TPAC2023

aphillips opened the pull request:
Address w3c/i18n-actions#58

Updating localizable-manifests to reflect our work with the app-manifest folks at TPAC and subsequent work on various formats.

This involves extensive additions to the spec.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/localizable-manifests/pull/16.html" title="Last updated on Nov 30, 2023, 4:11 PM UTC (26e7f78)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/localizable-manifests/16/7025e7c...aphillips:26e7f78.html" title="Last updated on Nov 30, 2023, 4:11 PM UTC (26e7f78)">Diff</a>

aphillips committed:
Address i18n-actions#58

Updating localizable-manifests to reflect our work with the app-manifest
folks at TPAC and subsequent work on various formats.

This involves extensive additions to the spec.

aphillips committed:
There are ~three~ er, uh, four solutions.

aphillips committed:
Update Addison's affiliation

aphillips committed:
Language negotiation and expert typo

aphillips committed:
Fix typos inherited from the explainer.

aphillips committed:
More work on the text

- Added links to JSON-LD and String-Meta definitions
- Made all of the examples examples.
- Added a note about the incompleteness of document defaults.

aphillips committed:
Additional clarification and improvements

xfq commented on index.html:
```diff
@@ -118,7 +123,12 @@ <h4>Document-Level Defaults</h4>
+			  <p>If your specification defines its own document level defaults, provide two optional fields:</p>
+			  
+			  <ul>
+				  <li>The document's default language field SHOULD be called <code>language</code> and SHOULD be specified to contain a valid [[BCP47]] language tag. Specifications SHOULD specify that implementations are only require to check if a [[BCP47]] language tag is "well-formed".</li>
```
Maybe we can add links to https://www.w3.org/TR/i18n-glossary/#dfn-valid and https://www.w3.org/TR/i18n-glossary/#dfn-well-formed here?

aphillips replied:
Done

aphillips committed:
Add examples of localized-but-not-translated

aphillips committed:
Add description of locale fallback and language tag matching

aphillips committed:
Tweaks and cleanup
- fix some spelling and grammar errors
- Move 6067/cldr link to a note
- other housekeeping

aphillips committed:
Link valid and well-formed to glossary

aphillips committed:
Fix link to well-formed
